packages/warehouses/src/warehouseClients/AthenaWarehouseClient.ts

```diff
@@ -0,0 +1,320 @@
+import {
+    CreateAthenaCredentials,
+    DimensionType,
+    Metric,
+    MetricType,
+    SupportedDbtAdapter,
+    WarehouseQueryError,
+} from '@lightdash/common';
+
+import {
+    AthenaClient,
+    GetQueryExecutionCommand,
+    GetQueryResultsCommand,
+    QueryExecutionState,
+    ResultSet,
+    StartQueryExecutionCommand,
+} from '@aws-sdk/client-athena';
+import { WarehouseCatalog } from '../types';
+import WarehouseBaseClient from './WarehouseBaseClient';
+
+export enum AthenaTypes {
+    BOOLEAN = 'boolean',
+    TINYINT = 'tinyint',
+    SMALLINT = 'smallint',
+    INTEGER = 'integer',
+    BIGINT = 'bigint',
+    REAL = 'real',
+    DOUBLE = 'double',
+    DECIMAL = 'decimal',
+    VARCHAR = 'varchar',
+    CHAR = 'char',
+    VARBINARY = 'varbinary',
+    JSON = 'json',
+    DATE = 'date',
+    TIME = 'time',
+    TIME_TZ = 'time with time zone',
+    TIMESTAMP = 'timestamp',
+    TIMESTAMP_TZ = 'timestamp with time zone',
+    INTERVAL_YEAR_MONTH = 'interval year to month',
+    INTERVAL_DAY_TIME = 'interval day to second',
+    ARRAY = 'array',
+    MAP = 'map',
+    ROW = 'row',
+    IPADDRESS = 'ipaddress',
+    UUID = 'uuid',
+}
+
+interface TableInfo {
+    database: string;
+    schema: string;
+    table: string;
+}
+
+const queryTableSchema = ({
+    database,
+    schema,
+    table,
+}: TableInfo) => `SELECT table_catalog
+                , table_schema
+                    , table_name
+                    , column_name
+                    , data_type
+            FROM ${database}.information_schema.columns
+            WHERE table_catalog = lower('${database}')
+                AND table_schema = lower('${schema}')
+                AND table_name = lower('${table}')
+            ORDER BY 1, 2, 3, ordinal_position`;
```

Comment:
this query is easy, but it's really really slow on scale. I've seen this query running ~2/3 minutes.
What we do in dbt-athena is to use glue apis. Also for example JDBC athena drivers uses the same concept, and in tools like Redash is possible to flag the usage of Glue APIs instead of using information_schema.

Reply:
@hhhonzik I'm happy to have a chat with you to change the behvior of the above, and make the retrival of metadata faster. The API to use should be simply glue `get_tables` https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue/client/get_tables.html for example in boto3. 

```
const getTables = (databaseName) => {
  const client = new GlueClient({});

  const command = new GetTablesCommand({
    DatabaseName: databaseName,
  });

  return client.send(command);
};
```
that supports only DatabaseName as filter, means that to to filter by a specific table we must do in memory.